### PR TITLE
Resource name tests and fixes

### PR DESCRIPTION
**Pull request by michaelbausor**
- Fix parse function for Any resource name type in Java
- Add test for any resource name in a response type
- Add test for any resource name in a page streaming response
- Do not generate paged response wrapper when no paged methods in Java
- Handle project arg in smoke tests with resource name oneofs in Java
- Remove unused method from ApiMethodTransformer
- Fix duplicated As in csharp

### CODE REVIEW

File: src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTransformer.java [hunk 1]
```diff
@@ -256,21 +268,25 @@ private StaticLangPagedResponseView generatePagedResponseWrapper(
...
+          namer.getPagedResponseIterateMethod(context.getFeatureConfig(), resourceFieldConfig);
+
+      String concreteResourceTypeName;
+      if (resourceFieldConfig.getResourceNameType() == ResourceNameType.ANY) {
+        concreteResourceTypeName = namer.publicClassName(Name.from("untyped_resource_name"));
+      } else {
+        concreteResourceTypeName = resourceTypeName;
+      }
+      String parseMethodName =
+          concreteResourceTypeName + "." + namer.publicMethodName(Name.from("parse"));
```
Comment (shinfan) [inline src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTransformer.java]: Move this to the JavaSurfaceNamer?

Comment (michaelbausor) [inline src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTransformer.java]: Fixed

**Comment by michaelbausor** [comment]
@garrettjonesgoogle @chrisdunelm PTAL

**Comment by garrettjonesgoogle** [comment]
LGTM, let Chris give his say too.

**Pull request merged**